spec/student_spec.rb

```diff
@@ -0,0 +1,23 @@
+require_relative './../student'
+require_relative './../person'
+require_relative './../decorators/nameable'
+
+describe 'Test student object' do
+  student = Student.new('A1', 18, 'Test Name', parent_permission: true)
+
+  it 'takes parameters and returns a Person object' do
+    expect(student).to be_kind_of(Person)
+  end
+
+  it 'takes parameters and returns a Nameable object' do
+    expect(student).to be_kind_of(Nameable)
+  end
+
+  it 'takes parameters and returns a Student object' do
+    expect(student).to be_instance_of(Student)
+  end
+
+  it "student name should be 'Test Name'" do
+    expect(student.name).to eql 'Test Name'
+  end
+end
```

Comment:
- [ ] Same here, you should have unit tests for all the methods in all the classes, Kindly make sure the `oplay_hooky and classroom` methods of the class Student have (at least) an 'it' block 👇🏻 check your person.rb file

```ruby
  def play_hooky
    '¯\(ツ)/¯'
  end

  def classroom=(classroom)
    @classroom = classroom
    classroom.students << self unless classroom.students.include?(self)
  end
```